src/components/Container.js

```diff
@@ -82,6 +82,49 @@ class Container extends React.PureComponent {
     } else return false
   }
 
+  renderStories() {
+    return this.props.stories.map((story, storyIndex) => {
+      const active = this.state.currentId === storyIndex
+
```

Comment:
Why should we have a different `renderStories()` function? The logic used by the `active` prop i.e. to show the currently active story is already implemented in the Container.js file. The `currentId` property defines the currently active story. Please use a similar approach for this implementation. 
You should access the current story by using `this.props.stories[this.state.currentId]`.

Reply:
The idea of using a different rendering for the stories is because we need to sync the progress with the horizontal animation. In the current architecture, I couldn't figure out a way to sync with just one story. For each story, it reset the animation and the progress, so it is always sync.

So I made a different implementation only when the use choose to have the horizontal animation.

Do you have any other idea on how to make this better? I can try.

Reply:
I tried another approach to solve the sync between animation and progress, by implementing a stories mapping with the active "status". It works for both the current implementation and the stories with animation. What do you think of this new approach?